packages/sanity/src/structure/documentActions/PublishAction.tsx

```diff
@@ -122,7 +125,10 @@ export const PublishAction: DocumentActionComponent = (props) => {
     return () => clearTimeout(timer)
   }, [changesOpen, publishState, hasDraft, onHistoryOpen])
 
+  const telemetry = useTelemetry()
+
   const handle = useCallback(() => {
+    telemetry.log(PublishDocument)
```

Comment:
@bjoerge Event should be called "Document Published" to follow convention. Could we also add an event property called "existing_document" <boolean> to see if it's a brand new document or an update to an existing one?

Reply:
@joneidejohnsen for a little bit of increased clarity, could it instead of `existing_document` be `new: true|false`?

Reply:
Actually, now that I think of it, should it mean:
- This document has not been published before
- Or "this is a new document" - e.g. created by the user just now and published straight away

Reply:
we can actually determine both, so we culd do this:
- `first_publish: boolean` - published for the first time (note that this will be true in case the document has previously been *un*published or deleted)
- `new: boolean` - created and published straight away

Reply:
Implemented it this way in https://github.com/sanity-io/sanity/commit/52ef1d222ab7bf5f5d1c6ca47b09620efb4d0897